tests/unit/DomUtils.js

```diff
@@ -0,0 +1,187 @@
+      DomUtils.addSelectOptions(select, options);
+      assert.lengthOf(select.options, 3, 'addSelectOptions should have added 3 options');
+    },
+
+    'addSelectOptionsWithPlaceholder': function() {
+      var select = domConstruct.create('select');
+      var options = {
+        data: [],
+        idProperty: 'id',
+        labelProperty: 'label',
+        placeholder: 'select placeholder'
+      };
+      DomUtils.addSelectOptionsWithPlaceholder(select, options);
+
+      // check if placeholder is set with empty data
+      assert.lengthOf(select.options, 1, 'addSelectOptionsWithPlaceholder should have added 1 option (placeholder)');
+      assert.strictEqual(select.options[select.selectedIndex].label, 'select placeholder',
+        'addSelectOptionsWithPlaceholder should have selected the placeholder option');
+
+      // check default placeholder
+      delete options.placeholder;
+      DomUtils.addSelectOptionsWithPlaceholder(select, options);
+      assert.lengthOf(select.options, 1, 'addSelectOptionsWithPlaceholder should have added 1 option (placeholder)');
+      assert.strictEqual(select.options[select.selectedIndex].label, 'Kies een optie...',
+        'addSelectOptionsWithPlaceholder should have selected the default placeholder option');
+
+      // check placeholder + data
+      options.placeholder = 'select with data';
+      options.data = [{id: 1, label: 'option 1'}, {id: 2, label: 'option 2'}, {id: 3, label: 'option 3'}];
+      DomUtils.addSelectOptionsWithPlaceholder(select, options);
+      assert.lengthOf(select.options, 4,
+        'addSelectOptionsWithPlaceholder should have added 4 options (data + placeholder)');
+      assert.strictEqual(select.options[select.selectedIndex].label, 'select with data',
+        'addSelectOptionsWithPlaceholder should have selected the default placeholder option');
+    },
+
+    'resetList': function() {
+      var ul = domConstruct.create('ul');
+      domConstruct.create('li', { innerHTML: 'item 1'});
+      domConstruct.create('li', { innerHTML: 'item 2'});
+      domConstruct.create('li', { innerHTML: 'item 3'});
+      domConstruct.create('li', { innerHTML: 'item 4'});
+
+      DomUtils.resetList(ul);
+      assert.lengthOf(ul.children, 0, 'resetList should have removed all list items');
+    },
+
+    'mergeArrays': function() {
+      var arr1 = [{id: 1, label: 'item 1'}, {id: 2, label: 'item 2'}, {id: 3, label: 'item 3'}];
+      var arr2 = [{id: 1, label: 'item 1'}, {id: 4, label: 'item 4'}, {id: 5, label: 'item 5'}];
+
+      // test with array of objects
+      assert.lengthOf(DomUtils.mergeArrays(arr1, arr2, 'id'), 5, 'mergeArrays should return an array with 5 items');
+      assert.sameDeepMembers(DomUtils.mergeArrays(arr1, arr2, 'id'), [{id: 1, label: 'item 1'},
+          {id: 2, label: 'item 2'}, {id: 3, label: 'item 3'}, {id: 4, label: 'item 4'}, {id: 5, label: 'item 5'}] ,
+        'mergeArrays should return a merged array with no duplicate elements');
```

Comment:
hier misscihen zeggen welke array er gecheckt wordt, anders weet je niet waar de fout gebeurt. of aparte suite voor gewone arrays en arrays met objecten.